### ISSUE TITLE
ERROR: failed to resolve runtime path : runtime  "io.containerd.kata-qemu.v2" "binary not installed "containerd-shim-kata-qemu-v2": file does not exist

### ISSUE THREAD

**Issue by bharath-manjunath** [kata-containers/kata-containers#10504]
**Facing problem installing Kata Containers using Kata-deploy**

Followed the below steps to install kata containers using kata deploy
```
$ kubectl apply -f https://raw.githubusercontent.com/kata-containers/kata-containers/main/tools/packaging/kata-deploy/kata-rbac/base/kata-rbac.yaml
$ kubectl apply -f https://raw.githubusercontent.com/kata-containers/kata-containers/main/tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml

##Installed runtime-classses
$ kubectl apply -f https://raw.githubusercontent.com/kata-containers/kata-containers/main/tools/packaging/kata-deploy/runtimeclasses/kata-runtimeClasses.yaml;
```

When i inspected containerd's config.toml i found the below content being created during installation
```
# vi /etc/containerd/config.toml

[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata-qemu]
runtime_type = "io.containerd.kata-qemu.v2"
runtime_path = "/opt/kata/bin/containerd-shim-kata-qemu-v2"
privileged_without_host_devices = true
pod_annotations = ["io.katacontainers.*"]
```

***Tried to run an example with kata-qemu***
```
$ kubectl apply -f https://raw.githubusercontent.com/kata-containers/kata-containers/main/tools/packaging/kata-deploy/examples/test-deploy-kata-qemu.yaml
```

**After following the above steps the pod got created but its in container-creating state as i'm encountering the below error when i described the test pod php-apache-kata-qemu created by running above command.**
```
 fatled to create containerd task: failed to start shim:  FailedCreatePodSandBox kubelet Failed to create pod sandbox: грс еггог: failed to resolve runtime path: runtime "to.containerd.kata-qemu.v2" "binary not installed "containerd-shim-kata-qemu-v2": file does not extst: 
```
![image (6)](https://github.com/user-attachments/assets/7cfdfaae-c1bb-49be-b62c-ae951dfb9f3d)

The corresponding binaries like containerd-shim-kata-quemu-v2 is not getting installed

**I'll be greatfull if someone can help me resolving this issue.**

**Comment by Apokleos** [comment]
@bharath-manjunath Could you please check and ensure the binary of `"/opt/kata/bin/containerd-shim-kata-qemu-v2"` is there ? and you might also check the `/opt/kata`

**Comment by bharath-manjunath** [comment]
> @bharath-manjunath Could you please check and ensure the binary of `"/opt/kata/bin/containerd-shim-kata-qemu-v2"` is there ? and you might also check the `/opt/kata`

I checked only /opt/kata/bin/containerd-shim-kata-v2 is getting installed by the above method but i can't find any binary installed specifically for qemu nor clh nor dragonball..!

**Comment by Apokleos** [comment]
> > @bharath-manjunath Could you please check and ensure the binary of `"/opt/kata/bin/containerd-shim-kata-qemu-v2"` is there ? and you might also check the `/opt/kata`
> 
> I checked it, only /opt/kata/bin/containerd-shim-kata-v2 is getting installed by the above method but i can't find any binary installed specifically for qemu nor clh nor dragonball..!
>
Yes, that is. maybe, something went wrong with kata-deploy. I think you can have a try again. Or using another way to address the current issue:
You can download the static tar from this site:  
`wget https://github.com/kata-containers/kata-containers/releases/download/3.10.1/kata-static-3.10.1-amd64.tar.xz`
and just untar this file: 
`tar -xvf kata-static-3.10.1-amd64.tar.xz`
then `move  ./opt/kata to /opt/` replacing the original one.
`mv ./opt/kata /opt/`

**Comment by bharath-manjunath** [comment]
> /opt/kata/bin/containerd-shim-kata-qemu-v2

@Apokleos the mentioned tar also **doesn't** have **/opt/kata/bin/containerd-shim-kata-qemu-v2 binary** it only has one generic binary **containerd-shim-kata-v2** 
![image](https://github.com/user-attachments/assets/5914ca38-9213-480e-91e7-4d0ccf599ac5)

**Comment by bharath-manjunath** [comment]
> > > @bharath-manjunath Could you please check and ensure the binary of `"/opt/kata/bin/containerd-shim-kata-qemu-v2"` is there ? and you might also check the `/opt/kata`
> > 
> > 
> > I checked it, only /opt/kata/bin/containerd-shim-kata-v2 is getting installed by the above method but i can't find any binary installed specifically for qemu nor clh nor dragonball..!
> 
> Yes, that is. maybe, something went wrong with kata-deploy. I think you can have a try again. Or using another way to address the current issue: You can download the static tar from this site: `wget https://github.com/kata-containers/kata-containers/releases/download/3.10.1/kata-static-3.10.1-amd64.tar.xz` and just untar this file: `tar -xvf kata-static-3.10.1-amd64.tar.xz` then `move ./opt/kata to /opt/` replacing the original one. `mv ./opt/kata /opt/`

@Apokleos the mentioned tar also **doesn't** have **/opt/kata/bin/containerd-shim-kata-qemu-v2 binary** it only has one generic binary **containerd-shim-kata-v2** 
![image](https://github.com/user-attachments/assets/5914ca38-9213-480e-91e7-4d0ccf599ac5)

**Comment by aki263** [comment]
+1 getting same error. Tried both firecracker vm and qemu vm's.

**Comment by aki263** [comment]
Adding below lines in `/etc/containerd/config.toml` solved the problem for me. 

```
[plugins."io.containerd.grpc.v1.cri".containerd]
snapshotter = "devmapper"
```

**Comment by bharath-manjunath** [comment]
**After making the below configuration changes it worked**

`[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata-qemu]
          runtime_type = "io.containerd.kata.v2"
          privileged_without_host_devices = true
          pod_annotations = ["io.katacontainers.*"]

          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata.options]
            ConfigPath = "/opt/kata/share/defaults/kata-containers/configuration-qemu.toml"`

**Comment by tingchienc** [comment]
Followed above configuration and copied the containerd-shim-kata-qemu-v2 to /usr/local/bin solve the issue.

This document help me understand how the runtime work with containerd and thus came up with the idea: https://github.com/containerd/containerd/tree/main/core/runtime/v2

**Comment by lbbym** [comment]
I've also met the same problem. and still not work with the config.
